### PR TITLE
ISOTPNativeSocket improvements

### PR DESCRIPTION
**Pull request by polybassa**
This PR update the ISOTPNativeSocket to store all arguments as internal variables for easier serialization. Also some docstrings to SuperSockets are added

### CODE REVIEW

File: scapy/supersocket.py [hunk 1]
```diff
@@ -116,14 +123,29 @@ def send(self, x):
     if WINDOWS:
         def _recv_raw(self, sock, x):
             # type: (socket.socket, int) -> Tuple[bytes, Any, Optional[float]]
-            """Internal function to receive a Packet"""
+            """Internal function to receive a Packet.
+
+            :param sock: Socket object from which data are received
+            :type sock: socket.socket
```
Comment (gpotter2) [inline scapy/supersocket.py:129]: I would argue that `:type:` and `:rtype:` shouldn't be used nowadays, considering we already include those types in type hints.

Sphinx and autodoc already parse the types from the type hints, and modern editors (like VS Code) for instance, also know how to get it. It really is duplicate information, and it's quite ugly IMHO.

Comment (polybassa) [reply scapy/supersocket.py]: I agree.

I've removed the type information

**Comment by polybassa** [comment]
@gpotter2 can this one get merged?

File: scapy/contrib/isotp/isotp_native_socket.py [hunk 2]
```diff
@@ -319,44 +319,63 @@ def __init__(self,
...
         )
 
-        self.__bind_socket(self.can_socket, self.iface, tx_id, rx_id)
-        self.ins = self.can_socket
-        self.outs = self.can_socket
-        if basecls is None:
-            log_isotp.warning('Provide a basecls ')
-        self.basecls = basecls
+        self.__bind_socket(can_socket, self.iface, self.tx_id, self.rx_id)
+        # make sure existing sockets are closed
```
Comment (gpotter2) [inline scapy/contrib/isotp/isotp_native_socket.py:367]: Hm.
I dislike having duplicate code. This is a duplicate of close, can't you just use it?

Comment (polybassa) [reply scapy/contrib/isotp/isotp_native_socket.py]: reusing close(), now!

**Review by gpotter2 — APPROVED** [review]

**Pull request merged**